go/adbc/driver/flightsql/utils.go

```diff
@@ -92,19 +93,37 @@ func adbcFromFlightStatusWithDetails(err error, header, trailer metadata.MD, con
 	// XXX: must check both headers and trailers because some implementations
 	// (like gRPC-Java) will consolidate trailers into headers for failed RPCs
 	for key, values := range header {
-		switch key {
-		case "content-type", "grpc-status-details-bin":
+		switch {
+		case key == "content-type":
+			// Not useful info
 			continue
+		case key == "grpc-status-details-bin":
+			// gRPC library parses this above via grpcStatus.Proto()
+			continue
+		case strings.HasSuffix(key, "-bin"):
+			for _, value := range values {
+				// that's right, gRPC stuffs binary data into a "string"
+				details = append(details, &adbc.BinaryErrorDetail{Name: key, Detail: []byte(value)})
+			}
 		default:
 			for _, value := range values {
 				details = append(details, &adbc.TextErrorDetail{Name: key, Detail: value})
 			}
 		}
```

Comment:
Could we leverage https://pkg.go.dev/google.golang.org/grpc@v1.62.0/status#Convert instead to get the message, error and details from the `err` instead of having to dig into the headers/trailers? Or is it not the same information?

Reply:
This is different. This is all the headers/trailers so we can reflect anything the server might send, not just the undocumented and partially-broken grpc-status-details-bin header that gRPC's Status covers.

Reply:
Also, we already call Convert above...